### PR TITLE
feat: Add console logger; add onto file logger

### PR DESCRIPTION
**Pull request by ebebbington**
Moves all the logger logic into this repo

Pulled in most of the logic drash had, but *did change a few things* which reflect the API

### CODE REVIEW

File: loggers/README.md [hunk 1]
```diff
@@ -4,48 +4,42 @@ A service to help log messages in the console or write logs to a file.
 
 ## Table of Contents
 
-- [ConsoleLogger](#consolelogger)
-- [FileLogger](#filelogger)
+- [Creating a Looger](#creating-a-logger)
```
Comment (crookse) [inline loggers/README.md]: what's a Looger lmao

Comment (ebebbington) [reply loggers/README.md]: Get with the times eric, it's the new name for it of course

Comment (ebebbington) [reply loggers/README.md]: https://github.com/drashland/services/pull/21/commits/f932a16c646be2afebba9da556aeb1913c6b441e

Comment (crookse) [reply loggers/README.md]: LOL

File: loggers/logger.ts [hunk 2]
```diff
@@ -0,0 +1,191 @@
+import { colors } from "../deps.ts";
+const encoder = new TextEncoder();
+
+export interface LoggerConfigs {
+  // deno-lint-ignore camelcase
+  tag_string?: string;
+  tag_string_fns?: { [key: string]: any }; // `any` because it can be a string, or an object with functions and/or strings, and the compiler throws errors when trying to execute certain logic with said type
+}
+
+type LogTypes = "info" | "debug" | "warn" | "error";
```
Comment (crookse) [inline loggers/logger.ts:10]: add `"trace"` and `"fatal"`

File: loggers/logger.ts [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+       * @param message - The message to be logged
+       *
+       * @returns Return the full logged message.
+       */
+  abstract error(message: string): string;
+
+  //////////////////////////////////////////////////////////////////////////////
+  // FILE MARKER - METHODS - PROTECTED /////////////////////////////////////////
+  //////////////////////////////////////////////////////////////////////////////
+
```
Comment (crookse) [inline loggers/logger.ts:90]: i think this base `Logger` class shouldn't know about how to log to the console or write to files. those should be in the console or file logger classes

File: loggers/logger.ts [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+    logType: LogTypes,
+    filename: string,
+  ): string {
+    const line = this.constructFullLogMessage(message, logType);
+    Deno.writeFileSync(filename, encoder.encode(line + "\n"), { append: true });
+    return line;
+  }
+
+  //////////////////////////////////////////////////////////////////////////////
+  // FILE MARKER - METHODS - ABSTRACT //////////////////////////////////////////
```
Comment (crookse) [inline loggers/logger.ts:127]: `FILE MARKER - METHODS - PRIVATE`

File: loggers/logger.ts [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+  //////////////////////////////////////////////////////////////////////////////
+
+  private constructFullLogMessage(message: string, logType: LogTypes): string {
```
Comment (crookse) [inline loggers/logger.ts:130]: need doc blocks breh

**Comment by ebebbington** [comment]
closing as we moved the logger stuff to unilogger

**Pull request closed without merging**